tools/build/Library/InstallUnpacker/classes/Download.php

```diff
@@ -0,0 +1,241 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+class PrestashopCouldNotDownloadLatestVersionException extends \Exception
+{
+}
+
+/**
+ * In charge of downloading the latest Prestashop Version.
+ *
+ * Most methods are copied from https://github.com/PrestaShop/autoupgrade/blob/master/classes/TaskRunner/Upgrade/Download.php
+ * and https://github.com/PrestaShop/autoupgrade/blob/master/classes/Tools14.php
+ */
+class Download
+{
+    const PRESTASHOP_API_RELEASES_XML_FEED = 'https://api.prestashop.com/xml/channel.xml';
+    const CACHED_FEED_FILENAME = 'XMLFeed';
+
+    /**
+     * @var BasicFileCache
+     */
+    private $cachingSystem;
+
+    /**
+     * @param BasicFileCache $cachingSystem optional FileCache
+     */
+    public function __construct(BasicFileCache $cachingSystem = null)
+    {
+        if (null === $cachingSystem) {
+            $cachingSystem = new BasicFileCache();
+        }
+
+        $this->cachingSystem = $cachingSystem;
+    }
+
+    /**
+     * @param string $source
+     * @param string $destination
+     *
+     * @return bool|int
+     */
+    public static function copy($source, $destination)
+    {
+        return @file_put_contents($destination, self::fileGetContents($source));
+    }
+
+    /**
+     * @param string $url
+     *
+     * @return bool|mixed|string
+     */
+    public static function fileGetContents($url)
+    {
+        $curl_timeout = 60;
+
+        if (!extension_loaded('openssl') and strpos('https://', $url) === true) {
+            $url = str_replace('https', 'http', $url);
+        }
+
+        $stream_context = null;
+        if (preg_match('/^https?:\/\//', $url)) {
+            $stream_context = @stream_context_create(array('http' => array('timeout' => $curl_timeout, 'header' => "User-Agent:MyAgent/1.0\r\n")));
+        }
+
+        if (in_array(ini_get('allow_url_fopen'), array('On', 'on', '1')) || !preg_match('/^https?:\/\//', $url)) {
+            $var = @file_get_contents($url, false, $stream_context);
+
+            if ($var) {
+                return $var;
+            }
```

Comment:
Should it need to return null here because of else statement below (Line 113)
Or is else statement useless and should return false?